version requirements for julia dependencies

cherrywoods opened the pull request:
I just added the versions of the dependencies that worked for me to the README file, as mentioned in issue #4. 

It would be nice to also add such version notices for the other components as well or even create a docker container that has all the right versions installed, I will see what I stumble across while working with this...

cherrywoods committed:
version requirements for julia dependencies

mykelk commented:
Awesome! Does this look good to you @kjulian3 ?

cherrywoods committed:
Updating Dockerfile

cherrywoods commented:
I added a Dockerfile that sets up Julia 1.1 and the required packages in a container now. The GenerateTables code is working, I didn't test the PolizyViz code.

zsunberg commented:
Would a Manifest.toml file make sense to manage the dependencies here?

On Fri, Mar 26, 2021 at 8:28 PM cherrywoods ***@***.***>
wrote:

> I added a Dockerfile that sets up Julia 1.1 and the required packages in a
> container now. The GenerateTables code is working, I didn't test the
> PolizyViz code.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sisl/HorizontalCAS/pull/5#issuecomment-808433810>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABALI2Z7ACBUVAAZXEJZNEDTFTHPTANCNFSM4XYRBIGQ>
> .
>

cherrywoods commented:
I think a Manifest.toml file would make sense. I have never really used Julia, so I didn't knew anything about it's package manager, but after what I read on it, a Manifest.toml file seems sensible. I will generate one, after I got to test the PolicyViz code as well. 

The docker container is I think still a good thing to have, because I also had some issues running this project with the latest Julia version, if I remember correctly and the best solution I found for using Julia 1.1 was using a container.

cherrywoods committed:
updating Dockerfile to run PolicyViz notebook

cherrywoods commented:
I didn't look into Manifest.toml yet, but I changed the Dockerfile to allow running the PolicyViz notebook inside the container. 
I also added a short paragraph at the end of the README file with instructions on running the notebook in the container. In the README I also added Revise to the dependency list, since it is imported in the notebook, but it might not be otherwise necessary.

cherrywoods committed:
Fixing docker container name to be all lowercase

cherrywoods committed:
Uniformly not using sudo in README Docker instructions

cherrywoods committed:
Enable running python scripts in docker container; Fix WebIO version.
